onedal/__init__.py

```diff
@@ -23,11 +24,20 @@
     import site
     import sys
 
+    plt_arch = platform.machine()
+    arch_dir = None
+    if plt_arch == "x86_64":
+        arch_dir = "intel64"
+    elif plt_arch == "aarch64":
+        arch_dir = "arm"
+    else:
+        arch_dir = plt_arch
+
```

Comment:
```suggestion
    arch_dir = platform.machine()
    plt_dict = {"x86_64":"intel64", "aarch64":"arm"}
    arch_dir = plt_dict[arch_dir] if arch_dir in plt_dict else arch_dir
```
More general for possible future expansion.